// grunt-json-to-object/Gruntfile.js
/*
 * grunt-json-to-object
 * https://github.com/letsmakesense/grunt-json-to-object
 *
 * Copyright (c) 2015 Peter Klein (Let's make sense GmbH)
 * Licensed under the MIT license.
 */

'use strict';

module.exports = function(grunt) {

  // Project configuration.
  grunt.initConfig({
    jshint: {
      all: [
        'Gruntfile.js',
        'tasks/*.js',
        '<%= nodeunit.tests %>'
      ],
      options: {
        jshintrc: '.jshintrc'
      }
    },

    // Before generating any new files, remove any previously-created files.
    clean: {
      tests: ['tmp']
    },

    // Configuration to be run (and then tested).
    json_to_object: {
      iso_langs: {
        options: {
          varname: 'languages'
        },
        files: {
          'tmp/iso_langs.js': ['test/fixtures/iso_langs.json']
        }
      },
      iso_langs_split: {
        options: {
          varname: 'languages_split'
        },
        files: {
          'tmp/iso_langs_split.js': [
            'test/fixtures/iso_langs_am.json',
            'test/fixtures/iso_langs_nz.json'
          ]
        }
      }
    },

    // Unit tests.
    nodeunit: {
      tests: ['test/*_test.js']
    }

  });

  // Actually load this plugin's task(s).
  grunt.loadTasks('tasks');

  // These plugins provide necessary tasks.
  grunt.loadNpmTasks('grunt-contrib-jshint');
  grunt.loadNpmTasks('grunt-contrib-clean');
  grunt.loadNpmTasks('grunt-contrib-nodeunit');

  // Whenever the "test" task is run, first clean the "tmp" dir, then run this
  // plugin's task(s), then test the result.
  grunt.registerTask('test', ['clean', 'json_to_object', 'nodeunit']);

  // By default, lint and run all tests.
  grunt.registerTask('default', ['jshint', 'test']);

};
